src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -206,28 +206,115 @@ internal object ReadODataOrCustomInstanceAnnotationValue(string annotationName,
         }
 
         /// <summary>
-        /// Check if a property value type in entity or complex value is deterministic .
+        /// Resolve the IEdmTypeReference of the current untyped value to be read.
         /// </summary>
         /// <param name="jsonReaderNodeType">The current JsonReader NodeType.</param>
         /// <param name="jsonReaderValue">The current JsonReader Value</param>
         /// <param name="payloadTypeName">The 'odata.type' annotation in payload.</param>
         /// <param name="payloadTypeReference">The payloadTypeReference of 'odata.type'.</param>
-        /// <returns>True if property value type is deterministic.</returns>
-        protected static bool IsKnownValueTypeForEntityOrComplex(JsonNodeType jsonReaderNodeType, object jsonReaderValue, string payloadTypeName, IEdmTypeReference payloadTypeReference)
+        /// <param name="primitiveTypeResolver">Function that takes a primitive value and returns an <see cref="IEdmTypeReference"/>.</param>
+        /// <param name="readUntypedAsString">Whether unknown properties should be read as a raw string value.</param>
+        /// <param name="generateTypeIfMissing">Whether to generate a type if not already part of the model.</param>
+        /// <returns>The <see cref="IEdmTypeReference"/> of the current value to be read.</returns>
+        [SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily")]

```

Comment:
It's complaining about multiple casts of jsonReaderValue to bool. However, any code path will only cast once.  If I had to replace the two casts with a single cast, and store the result, as suggested, then the cast would appear in code paths that didn't require the cast.

Added justification "Each code path casts to bool at most one time, and only if needed."

---
In reply to: [127302707](https://github.com/OData/odata.net/pull/894#discussion_r127302707) [](ancestors = 127302707)